_docs/tasks/security/authn-policy.md

```diff
@@ -0,0 +1,288 @@
+
+```
+cat <<EOF | istioctl create -n foo -f -
+apiVersion: "authentication.istio.io/v1alpha1"
+kind: "Policy"
+metadata:
+  name: "httpbin"
+spec:
+  targets:
+  - name: httpbin
+EOF
+```
+
+Re-run request from sleep.legacy, we should see success return code again (200), confirming service-level policy is overrule the namespace-level policy.
+
+```
+kubectl exec $(kubectl get pod -l app=sleep -n legacy -o jsonpath={.items..metadata.name}) -c sleep -n legacy -- curl http://httpbin.foo:8000/ip -s -o /dev/null -w "%{http_code}\n"
+```
+
+```
+200
+```
+
+
+## Setup end-user Authentication
+
+You will need a valid JWT (corresponding to the JWKS endpoint you want to use for the demo). Please follow the instruction [here](https://github.com/istio/istio/tree/master/security/tools/jwt) to create one. You can also use your own JWT/JWKS endpoint for the demo. Once you have that, let's export to some enviroment variables.
+
+
+```
+export JWKS=https://www.googleapis.com/service_accounts/v1/jwk/<YOUR-SVC-ACCOUNT>
+export TOKEN=<YOUR-TOKEN>
+```
+
+Also, for convenience, let expose httpbin.foo via ingress (for more details, see [ingress task]({{home}}/docs/tasks/traffic-management/ingress.html)).
+
+```
+cat <<EOF | kubectl apply -f -
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: httpbin-ingress
+  namespace: foo
+  annotations:
+    kubernetes.io/ingress.class: istio
+spec:
+  rules:
+  - http:
+      paths:
+      - path: /headers
+        backend:
+          serviceName: httpbin
+          servicePort: 8000
+EOF          
+```
+
+Get ingress IP
+```
+export INGRESS_HOST=$(kubectl get ing -n foo -o=jsonpath='{.items[0].status.loadBalancer.ingress[0].ip}')
+```
+
+And run test query
+```
+curl $INGRESS_HOST/headers -s -o /dev/null -w "%{http_code}\n"
+```
+
+```
+{
+  "headers": {
+    "Accept": "*/*",
+    "Content-Length": "0",
+    "Host": "35.230.123.105",
+    "User-Agent": "curl/7.58.0",
+    "X-B3-Sampled": "1",
+    "X-B3-Spanid": "d729acfa46c072ba",
+    "X-B3-Traceid": "d729acfa46c072ba",
+    "X-Envoy-Internal": "true",
+    "X-Ot-Span-Context": "d729acfa46c072ba;d729acfa46c072ba;0000000000000000",
+    "X-Request-Id": "8f232322-7c73-9470-8481-5ab64233adf9"
+  }
+}
+```
+
+Now, let's add a policy that require end-user JWT for httpbin.foo. If you follow previous section, the 'httpbin' authentication policy might areadly exist (run `kubectl get policies.authentication.istio.io -n foo` to confirm). To avoid create conflicting policies for the same service, we run istio replace for the same policy name (httpbin). Note in the example policy below, peer authentication (mTLS) is kept, but it can be removed independently from origin authencation settings.
```

Comment:
that require -> that requires
previous section -> the previous section
avoid create -> avoid creating
authencation -> authentication

Reply:
Done